configure

```diff
@@ -1013,6 +1019,9 @@ def configure_openssl(o):
   o['variables']['openssl_no_asm'] = 1 if options.openssl_no_asm else 0
   if options.use_openssl_ca_store:
     o['defines'] += ['NODE_OPENSSL_CERT_STORE']
+  if options.openssl_system_ca_path:
+    o['defines'] += ['NODE_OPENSSL_SYSTEM_CERT_PATH='
+        + options.openssl_system_ca_path]
```

Comment:
Operator on first line. :-)

I think you need the NODE_STRINGIFY macro because you pass a plain string but it should work without it if you wrap the string in quotes:
```python
o['defines'] += ['NODE_OPENSSL_SYSTEM_CERT_PATH="{}"'.format(options.openssl_system_ca_path)]
```
If you set it unconditionally, you don't need the `#if defined(...)` in node_crypto.cc.

Reply:
I like the suggestion and it works perfectly with for node_crypto.cc. I do run into an issue with `test-process-config.js` though:
```console
$ out/Release/node test/parallel/test-process-config.js
undefined:3
                       "defines": [ "NODE_OPENSSL_SYSTEM_CERT_PATH="/Users/danielbevenius/work/bucharest-gold/node-rpm/rpms/ca-bundle.crt""],
                                                                    ^

SyntaxError: Unexpected token / in JSON at position 164
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/Users/danielbevenius/work/nodejs/node/test/parallel/test-process-config.js:49:15)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Function.Module.runMain (module.js:682:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:613:3

```
I've tried all possible escaping combinations that I can think of in configure but they all seem to break the C++ code. 

I'll dig a little further but in the meantime I'll update with the operator on the correct line and also unconditionally set `NODE_OPENSSL_SYSTEM_CERT_PATH` and remove the `#if defined` in node_crypto.cc

Reply:
That's just a bug in the test.  It pretends that config.gyp is JSON (and tries to parse it as such) when it's not.

Reply:
Oh great, I'll see if I can come up with a fix for it. Thanks again.

Reply:
I've not found a good way of working/fixing the test. What I've suggested in https://github.com/nodejs/node/pull/16790/commits/f8980cd4c72b0851df2873800f8596bfdbc0c8ec is to skip the test if it cannot be parsed into valid JSON. Is this acceptable?

Reply:
Would this work?
```diff
diff --git a/test/parallel/test-process-config.js b/test/parallel/test-process-config.js
index 05616480e0..05f4ca915a 100644
--- a/test/parallel/test-process-config.js
+++ b/test/parallel/test-process-config.js
@@ -45,7 +45,9 @@ if (!fs.existsSync(configPath)) {
 let config = fs.readFileSync(configPath, 'utf8');
 
 // Clean up comment at the first line.
-config = config.split('\n').slice(1).join('\n').replace(/'/g, '"');
+config = config.split('\n').slice(1).join('\n');
+config = config.replace(/"/g, '\\"');
+config = config.replace(/'/g, '"');
 config = JSON.parse(config, function(key, value) {
   if (value === 'true') return true;
   if (value === 'false') return false;
```
Aside: looking at this again, I realize this sets the define build-wide, not just for files in src/.

`NODE_OPENSSL_CERT_STORE` does that too so it's okay for now but it's something we should fix sometime.

Reply:
Thanks, that worked like a charm. 

>Aside: looking at this again, I realize this sets the define build-wide, not just for files in src/.

Could you tell me or point me to an example of how I can limit them to only files in src?

Reply:
I couldn't find a proper example but the basic idea is that you restrict the define to a specific target:
```diff
diff --git a/node.gyp b/node.gyp
index 1c3e291471..6ca77a0c19 100644
--- a/node.gyp
+++ b/node.gyp
@@ -304,6 +304,14 @@
         # Warn when using deprecated V8 APIs.
         'V8_DEPRECATION_WARNINGS=1',
       ],
+
+      'conditions': [
+        [ 'openssl_system_ca_path!=""', {
+          'defines': [
+            'NODE_OPENSSL_SYSTEM_CERT_PATH="<(openssl_system_ca_path)"',
+          ],
+        }],
+      ],
     },
     {
       'target_name': 'mkssldef',
```
You can't see it but that's the `'target_name': '<(node_core_target_name)'` target.

Reply:
Ah I see, great. Thanks! 

I've updated inspired by your patch above. One thing I've done is [introduced](https://github.com/nodejs/node/pull/16790/commits/e4c77831e32bc6e9b052933fd1a6eb19e3d52fb2) a variable to the above target and setting it to an empty string if not set so that we always set `NODE_OPENSSL_SYSTEM_CERT_PATH`. Let me know what you think?